backend/src/main/java/ai/verta/modeldb/health/HealthServiceImpl.java

```diff
@@ -52,6 +52,10 @@
   private final Map<String, ServingStatus> statusMap = new ConcurrentHashMap<>();
   private static final Logger LOGGER = LogManager.getLogger(HealthServiceImpl.class);
 
+  public HealthServiceImpl() {
+    setStatus("", HealthCheckResponse.ServingStatus.SERVING);
```

Comment:
Will this automatically set the service as running even when it hasn't finished initialization? We have
```
healthStatusManager.setStatus("", HealthCheckResponse.ServingStatus.SERVING);
```
in the main already, so is this a problem of healthStatusManager not propagating to HealthServiceImpl? I want to make sure we're solving the base problem here as sometimes we'll set it as not serving (when there's an actual failure) and we need to make sure the system behaves appropriately in such scenarios.

Reply:
> Will this automatically set the service as running even when it hasn't finished initialization?
Yes, but it is never excess before it is up because spring boot do not allowed client request before it is completely start.

Before Main method restructured, we initialized `healthStatusManager.setStatus("", HealthCheckResponse.ServingStatus.SERVING);` in main method for rest endpoint as well as GRPC health check also and then after that same reference of  `healthStatusManager` is used while spring boot server will run. but now the spring boot server started before its initialization in main so we should have to initialize it again from its constructor.

I am sure it will not creating cause when actually backend is not in serving state.